setup/ci/codebuild.md

```diff
@@ -0,0 +1,121 @@
+You can edit your trust relationship to add the following policy.
+```
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": <Your Spinnaker Auth Role>
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+```
+
+## Configure Spinnaker to work with AWS CodeBuild
+
+Use the following Halyard commands to create an AWS CodeBuild account, enable the AWS CodeBuild integration, and re-deploy Spinnaker:
+```
+hal config ci codebuild account add $ACCOUNT_NAME \
+  --accountId $ACCOUNT_ID \
+  --assume-role $ASSUME_ROLE \
+  --region $REGION
+
+hal config ci codebuild enable
+
+hal deploy apply
+```
+
+## Configure an AWS CodeBuild stage
+
+To run an AWS CodeBuild build as part of a Spinnaker pipeline:
+
+1. Create a stage of type *AWS CodeBuild*.
+
+2. Configure the stage by selecting the AWS CodeBuild account to use to run the build, and selecting the project name
+from the dropdown list.
+
+3. (Optional) In the *Source Configuration* section, you might 
+  - Select the source artifact to use as build source. If not specified, the source configured in project is used.
+  - Specify the [source version](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_StartBuild.html#CodeBuild-StartBuild-request-sourceVersion)
+  of the build source.If not specified, the latest version is used.
+  - Specify the buildspec file. If left blank, the buildspec configured in project is used.
+  - Select secondary build sources. If not specified, the secondary sources configured in project is used
+
+4. (Optional) In the *Environment Configuration* section, you might specify the image tag or image digest that identifies the Docker image
+to use. If not specified, the image configured in project is used. **Note:** As prerequisite, follow this
+[user guide](https://docs.aws.amazon.com/codebuild/latest/userguide/sample-private-registry.html) to set up your registry credentials
+in project configuration.
+
+5. (Optional) In the *Advanced Configuration* section, you might specify environment variables to use in the build.
+Only environment variables in plain text are supported.
+
+6. (Optional) In the *Produces Artifacts* section, you might supply any artifacts that you expect the build to create in order to
+make these artifacts available to downstream stages. AWS CodeBuild supports creating S3 artifacts, which will be converted
+to Spinnaker artifacts and injected into the pipeline on completion of the build.
+
+While your build is executing, the stage details provide the following information
+* Current status of the build
+* ARN of the build
+* The link to view the build in the AWS CodeBuild Console
+* The link to view the build logs in AWS CloudWatch Logs
+* The link to view the build logs in S3
+
+## Configure your pipeline trigger (Optional)
+
+Configure your pipeline to be triggered by AWS CodeBuild build phase or state changes:
```

Comment:
Is this not required when the build is a stage in the pipeline?  Does Spinnaker poll for the completed build state while the stage is running a build?

Reply:
Spinnaker polls the build state by calling BatchGetBuilds every 10 seconds, so a codebuild trigger is not required.